File: data/readable_format/fortran/file1495.f90
program testProg
    implicit none
    integer :: i
    integer, dimension(2) :: nums
    integer :: numx
    integer, dimension(4) :: numse

    common /scr/ nums, numx
    equivalence (nums, numse(1)), (numx, numse(3))

    ! Main logic
    do i = 1, 4
        numse(i) = 10 + i
    end do

    ! Testing
    call testNums()
    call testNumse()
    call testNumx()

    print *, 'Fortran tests passed.'
end program testProg

subroutine testNums()
    integer, dimension(2) :: nums
    integer :: numx
    common /scr/ nums, numx
    integer expected(2)
    data expected /11, 12/
    if (all(nums == expected)) then
        print *, 'Nums test passed.'
    else
        print *, 'Nums test failed.'
    end if
end subroutine testNums

subroutine testNumse()
    integer, dimension(2) :: nums
    integer :: numx
    integer, dimension(4) :: numse
    common /scr/ nums, numx
    equivalence (nums, numse(1)), (numx, numse(3))
    integer expected(4)
    data expected /11, 12, 13, 14/
    if (all(numse == expected)) then
        print *, 'Numse test passed.'
    else
        print *, 'Numse test failed.'
    end if
end subroutine testNumse

subroutine testNumx()
    integer, dimension(2) :: nums
    integer :: numx
    common /scr/ nums, numx
    integer expected
    expected = 13
    if (numx == expected) then
        print *, 'Numx test passed.'
    else
        print *, 'Numx test failed.'
    end if
end subroutine testNumx

File: data/readable_format/cpp/file1495.cpp
#include <iostream>
#include <cassert>

struct SharedMemory {
    union {
        int nums[2];
        int numse[4];
    };
    int& numx = nums[1];
};

SharedMemory scr;

void simulateMainLogic() {
    for (int i = 0; i < 4; ++i) {
        scr.numse[i] = 10 + i + 1;
    }
}

void runTest() {
    int expectedNums[2] = {11, 12};
    int expectedNumse[4] = {11, 12, 13, 14};
    int expectedNumx = 12; // Correct expectation based on Fortran's behavior and memory layout

    simulateMainLogic();

    for (int i = 0; i < 2; ++i) {
        assert(scr.nums[i] == expectedNums[i]);
    }

    for (int i = 0; i < 4; ++i) {
        assert(scr.numse[i] == expectedNumse[i]);
    }

    assert(scr.numx == expectedNumx);

    std::cout << "C++ tests passed." << std::endl;
}

int main() {
    runTest();
    return 0;
}